modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/checkpoint/CheckpointHistory.java

```diff
@@ -49,6 +50,24 @@
  * This directory holds files for checkpoint start and end.
  */
 public class CheckpointHistory {
+    /** The message appears when no one checkpoint was not reserved for cache. */
+    private static final String NOT_RESERVED_WAL_REASON = "If checkpoint WAL history can't be reserved";
+
+    /** The message puts down to log when an exception happened during reading reserved WAL. */
+    private static final String WAL_SEG_CORRUPTED_REASON = "Segment corrupted";
+
+    /** Reason means no more history reserved for the cache. */
+    private static final String NO_MORE_HISTORY_REASON = "No more history for reservation";
+
+    /** Node does not have owning partitions. */
+    private static final String NO_PARTITIONS_OWNED_REASON = "No own partitions";
+
+    /** Reason means a checkpoint in history reserved can not be applied for cache. */
+    private static final String CHECKPOINT_NOT_APPLICABLE_REASON = "Checkpoint was marked as inapplicable for historical rebalancing";
+
+    /** That means all history reserved for cache. */
+    private static final String FULL_HISTORY_REASON = "Reserved full history for group";
```

Comment:
What does it mean? I don't get it

Reply:
This means we still have WAL, but all history from a moment when partitions were owned until current already presented.

Reply:
Full history were reserved